### PR TITLE
Add Shapes benchmarks for interactions

### PR DESCRIPTION
**Pull request by kevinyamauchi**
# Description
As discussed in #1562 , I have added a small benchmarking suite for two Shapes layer interactions we observed to be a bit slow (selecting shapes and dragging shapes). This PR adds:

- A benchmark for selecting a shape by simulating a click event
- A benchmark for dragging a shape (makes 5 drag event calls) by simulating a click and drag
- Updates the installation in the `asv.conf.json` to include the new qt installation flag (`[all]`)

The `ShapesInteractionSuite` results currently are:

```bash

(napari-dev) ➜  napari git:(add-shapes-benchmarks) ✗ asv dev -b ShapesInteractionSuite
· Discovering benchmarks
· Running 2 total benchmarks (1 commits * 1 environments * 2 benchmarks)
[  0.00%] ·· Benchmarking existing-py_Users_yamauc0000_miniconda3_envs_napari-dev_bin_python
[ 25.00%] ··· ...mark_shapes_layer.ShapesInteractionSuite.time_drag_shape                 ok
[ 25.00%] ··· ========== ==========
               n_shapes
              ---------- ----------
                  16      94.4±0ms
                  32      196±0ms
                  64      370±0ms
                 128      832±0ms
                 256      1.61±0s
              ========== ==========

[ 50.00%] ··· ...rk_shapes_layer.ShapesInteractionSuite.time_select_shape                 ok
[ 50.00%] ··· ========== ==========
               n_shapes
              ---------- ----------
                  16      54.7±0ms
                  32      112±0ms
                  64      401±0ms
                 128      812±0ms
                 256      1.02±0s
              ========== ==========
```

## Type of change
- [x] Bug-fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# References
This PR arises from discussion in #1562 

# How has this been tested?
- [x] I ran `asv dev -b ShapesInteractionSuite` to test the suite from my environment
- [x] I ran `asv continuous master -b ShapesInteractionSuite` to test the build
- [x] all tests pass with my change

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: asv.conf.json
Comment (kevinyamauchi) [inline asv.conf.json]: I had to add this to specify the qt option. I chose `[all]`, but I don't have a strong preference which backend we use.

This assumes that the config file (`asv.conf.json`) lives in the root directory where the installation instructions are. I think this is a safe assumption, but it would be nice to get some feedback if this is the best way to do this.

Comment (kevinyamauchi) [reply asv.conf.json]: Also, I modified this from this [asv docs](https://asv.readthedocs.io/en/stable/asv.conf.json.html?highlight=asv%20conf#build-command-install-command-uninstall-command). Does anybody know why there isn't a space between `python -m` and `pip`? It works, but it seems odd to me.

Comment (sofroniewn) [reply asv.conf.json]: @jni do you have thoughts here? Given that is from the docs I am fine with it

**Review by sofroniewn — APPROVED** [review]
These benchmarks looks great @kevinyamauchi. I'm approving now, but will see if @jni wants to weigh in on the install question about before merge. Feel free to merge tomorrow if you want.

File: asv.conf.json
Comment (jni) [reply asv.conf.json]: You can add a space if it makes you feel better. It would make *me* feel better too. =P But the GNU argument syntax is flexible, and if your argument has short form `a` and long form `option`, all of the following are valid:

- `-a argvalue`
- `-aargvalue`
- `--option argvalue`
- `--option=argvalue`

**Comment by jni** [comment]
Fantastic! @kevinyamauchi do you want to add that space and then we merge this? =)

File: asv.conf.json
Comment (kevinyamauchi) [reply asv.conf.json]: Ah TIL. Thanks for the explanation!

**Comment by kevinyamauchi** [comment]
>  @kevinyamauchi do you want to add that space and then we merge this? =)

Haha, yes. I'll do that right now.

**Comment by kevinyamauchi** [comment]
The tests have passed, so I will merge now. Thanks, everyone!

**Pull request merged**
